snuba/consumers/consumer.py

```diff
@@ -140,39 +163,48 @@ def close(self) -> None:
-        end_to_end_latency_sum = 0.0
+        snuba_latency_recorder = LatencyRecorder()
+        end_to_end_latency_recorder = LatencyRecorder()
+        sentry_received_latency_recorder = LatencyRecorder()
+
         for message in self.__messages:
             assert isinstance(message.value, BrokerValue)
+
             latency = write_finish - message.value.timestamp.timestamp()
-            latency_sum += latency
-            if max_latency is None or latency > max_latency:
-                max_latency = latency
-            if message.payload.origin_timestamp is not None:
-                end_to_end_latency = (
-                    write_finish - message.payload.origin_timestamp.timestamp()
+            snuba_latency_recorder.record(latency)
+
+            origin_timestamp = message.payload.origin_timestamp
+            if origin_timestamp is not None:
+                end_to_end_latency = write_finish - origin_timestamp.timestamp()
+                end_to_end_latency_recorder.record(end_to_end_latency)
+
+            sentry_received_timestamp = message.payload.sentry_received_timestamp
+            if sentry_received_timestamp is not None:
+                sentry_received_latency = (
+                    write_finish - sentry_received_timestamp.timestamp()
                 )
-                end_to_end_latency_sum += end_to_end_latency
-                if (
-                    max_end_to_end_latency is None
-                    or end_to_end_latency > max_end_to_end_latency
-                ):
-                    max_end_to_end_latency = end_to_end_latency
-
-        if max_latency is not None:
-            self.__metrics.timing("max_latency_ms", max_latency * 1000)
+                sentry_received_latency_recorder.record(sentry_received_latency)
+
+        if snuba_latency_recorder.max_ms:
+            self.__metrics.timing("max_latency_ms", snuba_latency_recorder.max_ms)
+            self.__metrics.timing("latency_ms", snuba_latency_recorder.avg_ms)
+        if end_to_end_latency_recorder.max_ms:
+            self.__metrics.timing(
+                "max_end_to_end_latency_ms", end_to_end_latency_recorder.max_ms
+            )
             self.__metrics.timing(
-                "latency_ms", (latency_sum / len(self.__messages)) * 1000
+                "end_to_end_latency_ms",
+                end_to_end_latency_recorder.avg_ms,
             )
-        if max_end_to_end_latency is not None:
+
+        if sentry_received_latency_recorder.max_ms:
             self.__metrics.timing(
-                "max_end_to_end_latency_ms", max_end_to_end_latency * 1000
+                "max_sentry_received_latency_ms",
+                sentry_received_latency_recorder.max_ms,
             )
             self.__metrics.timing(
-                "end_to_end_latency_ms",
-                (end_to_end_latency_sum / len(self.__messages)) * 1000,
+                "sentry_received_latency_ms",
+                sentry_received_latency_recorder.avg_ms,
```

Comment:
Why did we change the name of this metric? 
- There are teams using this as an SLO like profiling. 
- There may not be sentry in front of snuba. Thus it was called ned to end.